src/encoded/static/components/facets.js

```diff
@@ -0,0 +1,518 @@
+    facetLabel: PropTypes.string.isRequired,
+    selectedFacets: PropTypes.array.isRequired,
+    addGenomeFilter: PropTypes.func.isRequired,
+};
+
+export const Facet = (props) => {
+    const { facetTitle, facetName, facetArray, addGenomeFilter, selectedFacets } = props;
+    return (
+        <div className="facet">
+            <h4>{facetTitle}</h4>
+            <div className="facet-scrollable">
+                {Object.keys(facetArray).map((d) => {
+                    if (d === '') {
+                        return <hr />;
+                    }
+                    return <FacetButton
+                        selectedFacets={selectedFacets}
+                        buttonLabel={`${d} (${facetArray[d]})`}
+                        buttonName={d}
+                        facetLabel={facetName}
+                        addGenomeFilter={addGenomeFilter}
+                        key={d}
+                    />;
+                })}
+            </div>
+        </div>
+    );
+};
+
+Facet.propTypes = {
+    facetTitle: PropTypes.string.isRequired,
+    facetName: PropTypes.string.isRequired,
+    facetArray: PropTypes.object.isRequired,
+    addGenomeFilter: PropTypes.func.isRequired,
+    selectedFacets: PropTypes.array.isRequired,
+};
+
+const TypeaheadFacet = (props) => {
+    const { facetTitle, facetName, facetArray, handleSearch, addGenomeFilter, selectedFacets, unsanitizedSearchTerm } = props;
+    return (
+        <div className="facet">
+            <h4>{facetTitle}</h4>
+            <div className="typeahead-entry" role="search">
+                <i className="icon icon-search" />
+                <div className="searchform">
+                    <input type="search" aria-label={`Search to filter list of terms for ${facetName} facet`} placeholder="Search" value={unsanitizedSearchTerm} onChange={e => handleSearch(e, facetName)} name={`Search ${facetName} facet`} />
+                </div>
+            </div>
+            <div className="facet-scrollable">
+                <div className="top-shading hide-shading" />
+                <div
+                    className="term-list"
+                    onScroll={shadeOverflowOnScroll}
+                >
+                    {Object.keys(facetArray).map((d) => {
+                        if (d === '') {
+                            return <hr />;
+                        }
+                        return <FacetButton
+                            selectedFacets={selectedFacets}
+                            buttonLabel={`${d} (${facetArray[d]})`}
+                            buttonName={d}
+                            facetLabel={facetName}
+                            addGenomeFilter={addGenomeFilter}
+                            key={d}
+                        />;
+                    })}
+                </div>
+                <div className={`shading ${(facetArray.length < displayedTermsCount) ? 'hide-shading' : ''}`} />
+            </div>
+        </div>
+    );
+};
```

Comment:
Seems like `TypeaheadFacet` could either call `Facet` directly to reduce duplication, or rename it and have it take a `children` property, and use `Facet` as a child of this new component, again to reduce the code duplication.